implementations/micrometer-registry-azure/src/main/java/io/micrometer/azure/AzureMeterRegistry.java

```diff
@@ -0,0 +1,301 @@
+ * limitations under the License.
+ */
+package io.micrometer.azure;
+
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.StreamSupport.stream;
+
+import com.microsoft.applicationinsights.TelemetryClient;
+import com.microsoft.applicationinsights.TelemetryConfiguration;
+import com.microsoft.applicationinsights.telemetry.MetricTelemetry;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.DistributionSummary;
+import io.micrometer.core.instrument.FunctionCounter;
+import io.micrometer.core.instrument.FunctionTimer;
+import io.micrometer.core.instrument.Gauge;
+import io.micrometer.core.instrument.LongTaskTimer;
+import io.micrometer.core.instrument.Meter;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.TimeGauge;
+import io.micrometer.core.instrument.Timer;
+import io.micrometer.core.instrument.distribution.HistogramSnapshot;
+import io.micrometer.core.instrument.distribution.ValueAtPercentile;
+import io.micrometer.core.instrument.step.StepMeterRegistry;
+import io.micrometer.core.instrument.util.StringUtils;
+import io.micrometer.core.lang.Nullable;
+import java.util.Map;
+import java.util.Optional;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Publishes Metrics to Azure Application Insights
+ * @author Dhaval Doshi
+ */
+public class AzureMeterRegistry extends StepMeterRegistry {
+
+    /**
+     * TelemetryClient instance used to send metrics to Azure
+     */
+    private final TelemetryClient client;
+    private final Logger logger = LoggerFactory.getLogger(AzureMeterRegistry.class);
+
+    public AzureMeterRegistry(AzureConfig config, @Nullable TelemetryConfiguration configuration, Clock clock) {
+        this(config, clock, configuration, Executors.defaultThreadFactory());
+    }
+
+    private AzureMeterRegistry(AzureConfig config, Clock clock, @Nullable TelemetryConfiguration clientConfig, ThreadFactory threadFactory) {
+        super(config, clock);
+
+        config().namingConvention(new AzureNamingConvention());
+
+        if (clientConfig == null) {
+            // Get the current active instance of TelemetryConfiguration.
+            // This will only happen for Non SpringBoot scenario.
+            clientConfig = TelemetryConfiguration.getActive();
+        }
+
+        if (StringUtils.isEmpty(clientConfig.getInstrumentationKey())) {
+
+            // Pick Instrumentation Key From the Config if not set via instrumentationKey / Environment Variable
+            clientConfig.setInstrumentationKey(config.instrumentationKey());
+        }
+
+        requireNonNull(clientConfig);
+        requireNonNull(clientConfig.getInstrumentationKey());
+
+        this.client = new TelemetryClient(clientConfig);
```

Comment:
set SDK version for this client to ease troubleshooting of metric values. Also register it here https://github.com/Microsoft/ApplicationInsights-Home/blob/master/EndpointSpecs/SDK-VERSIONS.md

Reply:
@SergeyKanzhelev would you suggest having something like this: mm_java:version_numner where "mm" is the prefix ?

Reply:
Please let me know if the new proposed name in the change works. If yes I will make changes to the above link too.

Reply:
Prefix like `mm_` in proposed example is used to indicate platform/agent. Please use simply `mmj:version_number`

Reply:
Sure, mm:version_number makes sense to me. Will update it.